java/client/src/main/java/glide/api/models/BaseTransaction.java

```diff
@@ -3385,6 +3401,33 @@ public <ArgType> T xadd(
         return getThis();
     }
 
+    /**
+     * Adds an entry to the specified stream stored at <code>key</code>.<br>
+     * If the <code>key</code> doesn't exist, the stream is created.
+     *
+     * @implNote {@link ArgType} is limited to {@link String} or {@link GlideString}, any other type
+     *     will throw {@link IllegalArgumentException}.
+     * @see <a href="https://valkey.io/commands/xadd/">valkey.io</a> for details.
+     * @param key The key of the stream.
+     * @param values Field-value pairs to be added to the entry.
+     * @param options Stream add options {@link StreamAddOptions}.
+     * @return Command Response - The id of the added entry, or <code>null</code> if {@link
+     *     StreamAddOptionsBuilder#makeStream(Boolean)} is set to <code>false</code> and no stream
+     *     with the matching <code>key</code> exists.
+     */
+    public <ArgType> T xadd(
+            @NonNull ArgType key, @NonNull ArgType[][] values, @NonNull StreamAddOptions options) {
+        checkTypeOrThrow(key);
+        protobufTransaction.addCommands(
```

Comment:
If you add a protected `xadd`, you can reuse it in add public `xadd`. That will reduce number of UT without losing coverage

Reply:
I'm not sure I understand what you mean by adding a protected `xadd`. I didn't see any other examples in this file of this being done for other commands.